src/ray/common/cgroup/cgroup_setup.cc

```diff
@@ -40,6 +42,17 @@ const std::string kSystemCgroupFolder = []() {
 // Cgroup PID path for ray system components.
 const std::string kSystemCgroupProcs =
     ray::JoinPaths(kSystemCgroupFolder, "cgroup.procs");
+
+// Application cgroup related constants.
+const std::string kApplicationCgroupFolder = []() {
+  // Append UUID to application cgroup path to avoid conflict.
+  // Chances are that multiple ray cluster runs on the same filesystem.
+  return absl::StrFormat("/sys/fs/cgroup/ray_application_%s", GenerateUUIDV4());
```

Comment:
non blocking, is it possible to reuse the ray's session id here instead?

Reply:
Let me check how I could get session id at raylet main function.

Reply:
it's `cluster_id` in main.cc

Reply:
The UUID suffix comes from @jjyao 's previous comment: https://github.com/ray-project/ray/pull/48828#discussion_r1861065209, basically we're worrying if we have multiple clusters or jobs running on the same physical node (same shared filesystem thus), there could be conflict on cgroup.

A second thought, placing all processes into a single node-wise cgroup makes sense.

Reply:
I think cluster_id is exactly for this need

Reply:
if you have 2 ray nodes on a shared OS though - how do both of them get 0.8*total mem?

Reply:
> if you have 2 ray nodes on a shared OS though - how do both of them get 0.8*total mem?

That's exactly why I want to place all ray-related applications into one single cgroup, which doesn't have cluster-id or uuid suffix, just `/sys/fs/cgroup/ray_application_cgroup`, so whatever cluster or job you're at, all tasks / actors share the same max memory limitation.